pyloopkit/dose_entry.py

```diff
@@ -30,25 +30,31 @@ def net_basal_units(type_, value, start, end, scheduled_basal_rate, delivered_un
     scheduled basal is
     """
     MINIMUM_MINIMED_INCREMENT = 20
+    hours_ = hours(end, start)
+
+    scheduled_basal_units = scheduled_basal_rate * hours_
+
+    net_delivered_units = None
+    if delivered_units:
+        net_delivered_units = delivered_units - scheduled_basal_units
```

Comment:
Will this return the wrong answer for bolus doses? Scheduled basal shouldn't be subtracted from boluses.

Reply:
Ah, that is a really good question. I was interpreting this function as the net basal of any **insulin** delivered since all dose types are passed into it.

But tell me if I understand correctly, the purpose of the glucose_effects function is to get insulin net glucose effect (GE) of all insulin: `GE = GE_bolus + GE_temp_basal - GE_scheduled_basal`. And the code as I have it here is doing `GE = (GE_bolus - GE_scheduled_basal) + (GE_temp_basal - GE_scheduled_basal)`, which is effectively doubling counting the scheduled basal. Does that sound right?